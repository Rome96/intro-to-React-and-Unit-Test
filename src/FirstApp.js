import React from 'react'
import PropTypes from 'prop-types'

const FirstApp = ({greeting, subTitle}) => {
  // const greeting = 'Hola universo'
  return (
    <>
      <h1>{greeting}</h1>
      <p>{subTitle}</p>
    </>
  )
}

FirstApp.propTypes = {
  greeting: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  subTitle: 'Lorem ipsum dolor repudiandae debitis molestiae illo nobis dolorum esse Voluptate'
}

export default FirstApp