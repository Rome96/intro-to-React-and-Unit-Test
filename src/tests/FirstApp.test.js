import React from 'react'
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import FirstApp from '../FirstApp';

describe('Testing in component <FirstApp/>', () => {

  // test('FirstApp return Hola, soy Turiano', () => {
  //   const greeting = 'Hola, soy Turiano';
  //   const { getByText } = render(<FirstApp greeting={greeting} />);

  //   expect(getByText(greeting)).toBeInTheDocument()
  // });

  test('FirstApp return Hola, soy Turiano', () => {
    const greeting = "Hola, soy Turiano";
    const wrapper = shallow(<FirstApp greeting ={greeting}/>);
    expect(wrapper).toMatchSnapshot()
  });
  

});