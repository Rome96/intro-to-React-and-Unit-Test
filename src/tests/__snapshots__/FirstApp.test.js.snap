// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in component <FirstApp/> FirstApp return Hola, soy Turiano 1`] = `
<Fragment>
  <h1>
    Hola, soy Turiano
  </h1>
  <p>
    Lorem ipsum dolor repudiandae debitis molestiae illo nobis dolorum esse Voluptate
  </p>
</Fragment>
`;
